CMakeLists.txt

```diff
@@ -274,6 +271,16 @@ target_include_directories(onnx_proto PUBLIC
   $<INSTALL_INTERFACE:include>
   ${PROTOBUF_INCLUDE_DIRS})
 
+if (MSVC)
+  if (BUILD_SHARED_LIBS OR ONNX_BUILD_MAIN_LIB)
+    set(ONNX_API_DEFINE "-DONNX_API=__declspec(dllexport)")
+  else()
+    set(ONNX_API_DEFINE "-DONNX_API=")
+  endif()
+else()
+  set(ONNX_API_DEFINE "-DONNX_API=")
+endif()
+target_compile_definitions(onnx_proto PRIVATE ${ONNX_API_DEFINE})
```

Comment:
Setting this to `PRIVATE` will cause compilation issue for other code which used ONNX and include `onnx.pb.h`.